test/integration-tests/gas-tests/Seaport.gas.test.ts

```diff
@@ -2,18 +2,74 @@ import { CommandType, RoutePlanner } from '../shared/planner'
 import { UniversalRouter, Permit2 } from '../../../typechain'
 import snapshotGasCost from '@uniswap/snapshot-gas-cost'
 import { seaportOrders, seaportInterface, getAdvancedOrderParams } from '../shared/protocolHelpers/seaport'
-import { resetFork } from '../shared/mainnetForkHelpers'
-import { ALICE_ADDRESS, DEADLINE, OPENSEA_CONDUIT_KEY } from '../shared/constants'
+import { resetFork, WETH } from '../shared/mainnetForkHelpers'
+import { ALICE_ADDRESS, DEADLINE, MAX_UINT, OPENSEA_CONDUIT, OPENSEA_CONDUIT_KEY } from '../shared/constants'
+import { abi as ERC20_ABI } from '../../../artifacts/solmate/src/tokens/ERC20.sol/ERC20.json'
 import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers'
 import hre from 'hardhat'
 import deployUniversalRouter, { deployPermit2 } from '../shared/deployUniversalRouter'
+import { Contract } from 'ethers'
+import { getPermitSignature } from '../shared/protocolHelpers/permit2'
 const { ethers } = hre
 
 describe('Seaport Gas Tests', () => {
   let alice: SignerWithAddress
   let router: UniversalRouter
   let permit2: Permit2
   let planner: RoutePlanner
+  let weth: Contract
+
+  describe('ERC20 -> NFT', () => {
+    let bob: SignerWithAddress
+    beforeEach(async () => {
+      await resetFork(16635782)
+      // alice's permits fail w/ account not found so using bob from default signers
+      bob = (await ethers.getSigners())[1]
+      permit2 = (await deployPermit2()).connect(bob) as Permit2
+      router = (await deployUniversalRouter(permit2)).connect(bob) as UniversalRouter
+      planner = new RoutePlanner()
+      weth = new ethers.Contract(WETH.address, ERC20_ABI, bob)
+      const routerSigner = await ethers.getImpersonatedSigner(router.address)
+      // alice deposits 10 eth into weth
+      await bob.sendTransaction({ to: weth.address, value: ethers.utils.parseEther('10') })
+      // approve permit2 for all for alice's weth
+      await weth.connect(bob).approve(permit2.address, ethers.constants.MaxUint256)
+      // custom approve the conduit key for router
+      await weth.connect(routerSigner).approve(OPENSEA_CONDUIT, MAX_UINT)
+    })
+
+    it('gas: fufillAdvancedOrder with ERC20', async () => {
```

Comment:
would be great to see the same things in this file
- putting the other tests in an ETH->NFT block
- a test without the approve command